src/FreeAge/client/render_window.cpp

```diff
@@ -2289,25 +2290,23 @@ void RenderWindow::RenderSelectionPanel(QOpenGLFunctions_3_2_Core* f) {
     
     // Display the object's HP
     if (singleSelectedObject->GetHP() > 0) {
-      u32 maxHP;
-      if (singleSelectedObject->isUnit()) {
-        maxHP = GetUnitMaxHP(AsUnit(singleSelectedObject)->GetType());
+      int maxHP = singleSelectedObject->GetObjectStats().maxHp;
```

Comment:
Probably more places in the code could use the `GetObjectStats()` to reduce code duplication, I didn't search for them.